src/mbgl/annotation/annotation_tile.hpp

```diff
@@ -45,7 +45,10 @@ class AnnotationTileLayer : public GeometryTileLayer {
     AnnotationTileLayer(std::string);
 
     std::size_t featureCount() const override { return features.size(); }
-    std::unique_ptr<GeometryTileFeature> getFeature(std::size_t i) const override { return std::make_unique<AnnotationTileFeature>(features[i]); }
+    std::unique_ptr<GeometryTileFeature> getFeature(std::size_t i) const override {
+        assert(i < features.size());
+        return std::make_unique<AnnotationTileFeature>(features[i]);
```

Comment:
For logic errors, it's better to crash -- crashes are picked up by reporting tools, we get a bug report, and can fix the issue. In comparison, logs are mostly ignored.